### ISSUE TITLE
[Android] [Regression] CarouselView throws unhandled java.lang.RuntimeException paging through images - Canvas: trying to use a recycled bitmap

### ISSUE THREAD

**Issue by mfeingol** [dotnet/maui#13182]
### Description

My app includes a CarouselView that can be used to scroll through images found on Flickr and displayed full screen from direct download URLs. Using the CarouselView, the user can page through images. By around the seventh image or so, I observe this exception in the device monitor:

```
java.lang.RuntimeException: Canvas: trying to use a recycled bitmap android.graphics.Bitmap@72c6291
	at android.graphics.BaseCanvas.throwIfCannotDraw(BaseCanvas.java:66)
	at android.graphics.RecordingCanvas.throwIfCannotDraw(RecordingCanvas.java:277)
	at android.graphics.BaseRecordingCanvas.drawBitmap(BaseRecordingCanvas.java:88)
	at android.graphics.drawable.BitmapDrawable.draw(BitmapDrawable.java:548)
	at android.widget.ImageView.onDraw(ImageView.java:1436)
	at android.view.View.draw(View.java:22350)
	at android.view.View.updateDisplayListIfDirty(View.java:21226)
	at android.view.View.draw(View.java:22081)
	at android.view.ViewGroup.drawChild(ViewGroup.java:4516)
	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4277)
	at android.view.View.updateDisplayListIfDirty(View.java:21217)
	at android.view.View.draw(View.java:22081)
	at android.view.ViewGroup.drawChild(ViewGroup.java:4516)
	at androidx.recyclerview.widget.RecyclerView.drawChild(RecyclerView.java:5204)
	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4277)
	at android.view.View.draw(View.java:22353)
	at androidx.recyclerview.widget.RecyclerView.draw(RecyclerView.java:4603)
	at android.view.View.updateDisplayListIfDirty(View.java:21226)
	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4500)
	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4473)
	at android.view.View.updateDisplayListIfDirty(View.java:21186)
	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4500)
	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4473)
	at android.view.View.updateDisplayListIfDirty(View.java:21186)
	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4500)
	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4473)
	at android.view.View.updateDisplayListIfDirty(View.java:21186)
	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4500)
	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4473)
	at android.view.View.updateDisplayListIfDirty(View.java:21186)
	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4500)
	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4473)
	at android.view.View.updateDisplayListIfDirty(View.java:21186)
	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4500)
	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4473)
	at android.view.View.updateDisplayListIfDirty(View.java:21186)
	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4500)
	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4473)
	at android.view.View.updateDisplayListIfDirty(View.java:21186)
	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4500)
	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4473)
	at android.view.View.updateDisplayListIfDirty(View.java:21186)
	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4500)
	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4473)
	at android.view.View.updateDisplayListIfDirty(View.java:21186)
	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4500)
	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4473)
	at android.view.View.updateDisplayListIfDirty(View.java:21186)
	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4500)
	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4473)
	at android.view.View.updateDisplayListIfDirty(View.java:21186)
	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4500)
	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4473)
	at android.view.View.updateDisplayListIfDirty(View.java:21186)
	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4500)
	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4473)
	at android.view.View.updateDisplayListIfDirty(View.java:21186)
	at android.view.ThreadedRenderer.updateViewTreeDisplayList(ThreadedRenderer.java:559)
	at android.view.ThreadedRenderer.updateRootDisplayList(ThreadedRenderer.java:565)
	at android.view.ThreadedRenderer.draw(ThreadedRenderer.java:642)
	at android.view.ViewRootImpl.draw(ViewRootImpl.java:4101)
	at android.view.ViewRootImpl.performDraw(ViewRootImpl.java:3828)
	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:3099)
	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1952)
	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:8171)
	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:972)
	at android.view.Choreographer.doCallbacks(Choreographer.java:796)
	at android.view.Choreographer.doFrame(Choreographer.java:731)
	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:957)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:223)
	at android.app.ActivityThread.main(ActivityThread.java:7656)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
```

There's not much on the .NET side:

```
 	0xFFFFFFFFFFFFFFFF in Android.Runtime.JNIEnv.monodroid_debugger_unhandled_exception	C#
 	0x59 in Android.Runtime.JNIEnv.PropagateUncaughtException at /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Android.Runtime/JNIEnv.cs:308,5	C#
```

This used to work fine in Xamarin.Forms.

### Steps to Reproduce

```
        <CarouselView ItemsSource="{Binding Images}" BackgroundColor="Black" Loop="False"
                      CurrentItem="{Binding SelectedImage, Mode=TwoWay}">
            <CarouselView.ItemTemplate>
                <DataTemplate x:DataType="xxx">
                    <Image Source="{Binding ImageUrl}" />
                </DataTemplate>
            </CarouselView.ItemTemplate>
        </CarouselView>
```

### Link to public reproduction project repository

Let me know if one is needed.

### Version with bug

7.0 (current)

### Last version that worked well

7.0 (current)

### Affected platforms

Android

### Affected platform versions

11 and 13

### Did you find any workaround?

No.

**Comment by PureWeen** [comment]
This should be fixed by the upcoming release of MAUI

if it still isn't fixed, please let us know and we can reopen the issue.

**Comment by hartez** [comment]
#12310 was the fix.

**Comment by mfeingol** [comment]
Verified it's fixed with VS 17.5.0 Preview 6.0. Thanks.
